views/layouts/MainLayout.jsx

```diff
@@ -0,0 +1,61 @@
+import React, { useState, useEffect, useContext } from 'react'
+import { Outlet, useLocation, useNavigate } from 'react-router-dom'
+
+import Header from '../components/Header'
+import Sidebar from '../components/Sidebar'
+
+import getAvatar from '../fe-utils/avatarUtil'
+import getCounts from '../fe-utils/countUtil'
+import { fetchSubjects } from '../fe-utils/fetchUtil'
+import { AuthContext } from '../contexts/AuthContext'
+import { headerTitle } from './helpers'
+
+const MainLayout = ({ classes, theme }) => {
+  const [user] = useContext(AuthContext)
+  const { pathname } = useLocation()
+  const navigate = useNavigate()
+
+  const [openDrawer, setOpenDrawer] = useState(false)
+  const [sideBarState, setSideBarState] = useState({
+    totalDays: 0,
+    totalStudies: 0,
+    totalSubjects: 0,
+  })
+  const [avatar, setAvatar] = useState('')
```

Comment:
we don't need state for this.

```suggestion
  const avatar = getAvatar({ user })
```

Reply:
[844ab7b](https://github.com/AMP-SCZ/dpdash/pull/317/commits/844ab7b0ea6f0f754f676e4f6a40563d84b43377)